src/main/java/com/salesforce/mirus/KafkaMonitor.java

```diff
@@ -217,6 +269,8 @@ boolean partitionsChanged() {
     Map<Boolean, List<TopicPartition>> partitionedSourceIds =
         sourcePartitionList
             .stream()
+            .map(this::applyTopicNameConfigsToTopic)
+            .map(this::applyRoutersToTopic)
```

Comment:
I kind of feel that prefix and suffix should be added _after_ the transformation router has executed, not before, but I'm not sure that's anything more than a personal opinion--and in any case, it should match what is already happening in the other part of the code. @pdavidson100 what do you think?

Reply:
I agree that's the more logical way, but [Kafka Connect applies transformations after the SourceTask finishes processing](https://github.com/apache/kafka/blob/af634a4a98eaa2457752e3f2841720020e0e9ad0/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java#L218) each batch of records. Transformations are applied inside of `WorkerSourceTask::sendRecords()`, which is only called after `MirusSourceTask::poll()` is called from `WorkerSourceTask::poll()`, so the Mirus Connector and Task aren't able to affect records after they're transformed.

#31 suggests deprecating the prefix/suffix options entirely since RegexRouter supports that fairly easily, which I think the best long-term plan.

Reply:
Because this is a Source Connector the transformations are applied _after_ we return the records to the framework, so necessarily after the topic name transformation happens in the connector code.  I think @parafiend has the order right.